src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/EnvelopeEventProcessorTest.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services;
+
+import com.microsoft.azure.servicebus.IMessage;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.bulkscanprocessorclient.client.BulkScanProcessorClient;
+
+import java.util.concurrent.ExecutionException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EnvelopeEventProcessorTest {
+
+    @Mock
+    BulkScanProcessorClient bulkScanProcessorClient;
+
+    private EnvelopeEventProcessor envelopeEventProcessor;
+
+    @Mock
+    private IMessage someMessage;
+    private static final String MSG_ID = "hello!";
+
+    @Before
+    public void before() {
+        envelopeEventProcessor = new EnvelopeEventProcessor(bulkScanProcessorClient);
+    }
+
+    @Test
+    public void should_use_queue_message_id_to_read_envelope() throws ExecutionException, InterruptedException {
+        //given
+        given(someMessage.getMessageId()).willReturn(MSG_ID);
+
+        // when
+        envelopeEventProcessor.onMessageAsync(someMessage).get();
+
+        // then
+        ArgumentCaptor<String> argumentCaptor = ArgumentCaptor.forClass(String.class);
+        verify(bulkScanProcessorClient).getEnvelopeById(argumentCaptor.capture());
+        assertThat(argumentCaptor.getValue()).isEqualTo(MSG_ID);
+    }
```

Comment:
a test for what happens if `bulkScanProcessorClient` throws an exception would be nice

Reply:
Sure ... Ill add the try catch that I commented on too.